modules/core/src/v2/coins/dot.ts

```diff
@@ -0,0 +1,249 @@
+   * @returns {Boolean} is it valid?
+   */
+  isValidPrv(prv: string): boolean {
+    return accountLib.Dot.Utils.default.isValidPrivateKey(prv);
+  }
+
+  /**
+   * Return boolean indicating whether input is valid public key for the coin
+   *
+   * @param {String} address the pub to be checked
+   * @returns {Boolean} is it valid?
+   */
+  isValidAddress(address: string): boolean {
+    return accountLib.Dot.Utils.default.isValidAddress(address);
+  }
+
+  /**
+   * Explain/parse transaction
+   * @param params
+   * @param callback
+   */
+  explainTransaction(
+    params: ExplainTransactionOptions,
+    callback?: NodeCallback<any>
+  ): Bluebird<any> {
+    throw new MethodNotImplementedError('Dot recovery not implemented');
+  }
+
+  verifySignTransactionParams(params: SignTransactionOptions): VerifiedTransactionParameters {
+    const prv = params.prv;
+    const addressVersion = params.txPrebuild.addressVersion;
+
+    const txHex = params.txPrebuild.txHex;
+
+    if (_.isUndefined(txHex)) {
+      throw new Error('missing txPrebuild parameter');
+    }
+
+    if (!_.isString(txHex)) {
+      throw new Error(`txPrebuild must be an object, got type ${typeof txHex}`);
+    }
+
+    if (_.isUndefined(prv)) {
+      throw new Error('missing prv parameter to sign transaction');
+    }
+
+    if (!_.isString(prv)) {
+      throw new Error(`prv must be a string, got type ${typeof prv}`);
+    }
+
+    if (!_.has(params.txPrebuild, 'key')) {
+      throw new Error('missing public key parameter to sign transaction');
+    }
+
+    if (!_.isNumber(addressVersion)) {
+      throw new Error('missing addressVersion parameter to sign transaction');
+    }
+
+    // if we are receiving addresses do not try to convert them
+    const signer = !accountLib.Dot.Utils.default.isValidAddress(params.txPrebuild.key)
+      ? new accountLib.Dot.KeyPair({ pub: params.txPrebuild.key }).getAddress()
+      : params.txPrebuild.key;
+    return { txHex, addressVersion, prv, signer };
+  }
+
+  /**
+   * Assemble keychain and half-sign prebuilt transaction
+   *
+   * @param params
+   * @param params.txPrebuild {TransactionPrebuild} prebuild object returned by platform
+   * @param params.prv {String} user prv
+   * @param callback
+   * @returns {Bluebird<SignedTransaction>}
+   */
+  signTransaction(
+    params: SignTransactionOptions,
+    callback?: NodeCallback<SignedTransaction>
+  ): Bluebird<SignedTransaction> {
+    const self = this;
+    return co<SignedTransaction>(function* () {
+      const { txHex, signer, prv } = self.verifySignTransactionParams(params);
+      const factory = accountLib.register(self.getChain(), accountLib.Dot.TransactionBuilderFactory);
+      const txBuilder = factory.from(txHex);
+      txBuilder
+        .validity(params.txPrebuild.validity)
+        .referenceBlock(params.txPrebuild.referenceBlock)
+        .version(params.txPrebuild.version)
+        .sender({ address: signer })
```

Comment:
Are these parameters always provided? Or are they optional? If they are optional, we should validate they are present (and valid) before assigning them to the builder

Reply:
They're always mandatory. Wrote a test to ensure that.